python_on_whales/components/system/cli_wrapper.py

```diff
@@ -57,9 +67,42 @@ def disk_free(self) -> DiskFreeResult:
         full_cmd = self.docker_cmd + ["system", "df", "--format", "{{json .}}"]
         return DiskFreeResult(run(full_cmd))
 
-    def events(self):
-        """Not yet implemented"""
-        raise NotImplementedError
+    def events(
+        self,
+        since: Union[None, datetime.datetime, datetime.timedelta] = None,
+        until: Union[None, datetime.datetime, datetime.timedelta] = None,
+        filters: Union[None, Dict[str, str]] = None,
+    ) -> Iterator[DockerEvent]:
+        """Return docker events information up to the current point in time.
+
+        # Arguments
+            since:  Show all events created since timestamp
+            until: 	Stream events until this timestamp
+            filters: See the [Docker documentation page about filtering
+                ](https://docs.docker.com/engine/reference/commandline/events/#filtering).
+        # Returns
+            A iterator which will yield DockerEvent objects from stdout/stderr
+
+        [reference page for
+        system events](https://docs.docker.com/engine/api/v1.40/#operation/SystemEvents)
+        """
+        full_cmd = self.docker_cmd + [
+            "system",
+            "events",
+            "--format",
+            "{{json .}}",
+        ]
+        full_cmd.add_simple_arg("--since", format_time_arg(since))
+        full_cmd.add_simple_arg("--until", format_time_arg(until))
+        full_cmd.add_args_list("--filter", format_dict_for_cli(filters))
+        iterator = stream_stdout_and_stderr(full_cmd)
+        event_function = (
```

Comment:
I believe using the `yield` keyword would make the code easier to read. You can do something like this:
```python
for stream_origin, stream_content in stream_stdout_and_stderr(...):
    if stream_origin == "stdout":
        yield DockerEvent.parse_raw(stream_content)
```

Reply:
done